### PR TITLE
Reduce difficulty of collection quests

### LINKED ISSUE #5618: Eternal Quest - Collection quests are too slow
Our party has spent over two months collecting Testimonies for The Golden Knight Chain Part 2. It is still not done.

Is this normal? Morale is low... :(
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/24348254-eternal-quest-collection-quests-are-too-slow?utm_campaign=plugin&utm_content=tracker%2F68393&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F68393&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by SabreCat**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #5618. Combines PRs #7987 and #8392. @paulwasit / @Alys, if you'd prefer to continue work on your own PRs, just say so!

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

>Reduce the number of required items for collection quests, with values proposed by [lemoness](https://github.com/HabitRPG/habitrpg/issues/5618#issuecomment-157479872):

| Quest Name | Quest Object Name | Old value(s) | new value(s) |
| :-- | :-- | :-- | :-- |
| Dish Disaster | atom1 | 20 | no change |
| Find the Cub | evilsanta2 | 30 (10 branches + 20 tracks) | no change |
| Find the Lair of the Wyrm | vice2 | 45 | 30 |
| Egg Hunt | egg | 100 | 40 |
| Message in a Bottle | dilatoryDistress1 | 50 (25 fireCoral + 25 blueFins) | 40 (20 fireCoral + 20 blueFins) |
| A Stern Talking-To | goldenknight1 | 300 | 60 |
| The Moonstone Chain | moonstone1 | 500 | 100 |

---

UUID: b6d7a7ca-98e1-4f11-b895-47c2b3667b9d

### CODE REVIEW

**Comment by SabreCat** [comment]
Gave @Alys's migration script a go locally; it worked fine! It's a little different style than other migrations we've got, but what's one more type of migration in the pile, eh? 😛

File: migrations/20170111_announce_collection_quest_change_in_parties.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  throw err;
+}
+
+function findPartiesWithCollectionQuest () {
+  logger.info('Looking up groups on collection quests...');
+
+  return Groups.find({'quest.key': {$in: COLLECTION_QUESTS }}, ['name','quest']).toArray().then((groups) => {
+    logger.success('Found', groups.length, 'parties on collection quests');
+
+    let groupsOfInterest = groups.reduce((array, party) => {
```
Comment (paglias) [inline migrations/20170111_announce_collection_quest_change_in_parties.js]: why not return groups directly?

File: migrations/20170111_announce_collection_quest_change_in_parties.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  var newMessage = {
+    'id' : uuid.v4(),
+    'text' : message,
+    'timestamp': Date.now(),
+    'likes': {},
+    'flags': {},
+    'flagCount': 0,
+    'uuid': 'system'
+  };
+  return Groups.findOneAndUpdate({_id: group._id}, {$push:{"chat" :{$each: [newMessage], $position:0}}}, {returnOriginal: false});
```
Comment (paglias) [inline migrations/20170111_announce_collection_quest_change_in_parties.js:87]: .update instead of findOneAndUpdate should work too but not a big deal

**Review by paglias — DISMISSED** [review]

File: migrations/20170111_announce_collection_quest_change_in_parties.js (hunk 2 above)
Comment (SabreCat) [reply migrations/20170111_announce_collection_quest_change_in_parties.js:87]: I played with that a bit, but the returned data on `update` is structured differently than `findOneAndUpdate`, in such a way that it made a hash of the ensuing `logger` calls.

Comment (paglias) [reply migrations/20170111_announce_collection_quest_change_in_parties.js:87]: 👍

**Comment by Alys** [comment]
@SabreCat Thanks very much for this! I'm sorry about the migration script format.  I'd copied it from one of the existing scripts but not from a recent one. :(  It was one of the convert_quest_collection ones I think. It might be worth putting a comment at the top of this new script to say that it shouldn't be copied in future, to stop this method breeding further.

File: migrations/20170111_announce_collection_quest_change_in_parties.js (hunk 2 above)
Comment (SabreCat) [reply migrations/20170111_announce_collection_quest_change_in_parties.js:87]: Is this OK to merge, then? Or are there still changes needed?

**Comment by SabreCat** [comment]
It's OK, it'll get the job done. We should probably muck out the `migrations` folder in some fashion eventually, heh...

**Comment by Alys** [comment]
+1 for mucking out the migrations folder. "The Augean Stables project". :slightly_smiling_face:  We could use the standard method for dealing with a backlog: move everything to a subdirectory called "old" except for the scripts we **know** are current and good, then as we find old ones that we need, move them back. It would create a bit of a mess with commit history (although not so bad I think because I'm pretty sure git handles file moves well) but it would certainly neaten the directory rapidly. But it's up to you of course! You're the one who uses it most.

**Review by TheHollidayInn — APPROVED** [review]
Looks good!

**Pull request merged**